lib/specValidator.js

```diff
@@ -51,610 +178,508 @@ class SpecValidator {
+          self.specValidationResult.validateSpec.error = e;
+          log.error(Constants.Errors);
+          log.error('------');
+          self.updateValidityStatus();
+          log.error(e);
+        } else {
+          self.specValidationResult.validateSpec.result = `The spec ${self.specPath} is sematically valid.`
+        }
+        if (validationResult.warnings && validationResult.warnings.length > 0) {
+          self.specValidationResult.validateSpec.warning = validationResult.warnings;
+          log.warn(Constants.Warnings);
+          log.warn('--------');
+          log.warn(util.inspect(validationResult.warnings));
+        }
+      }
+    } catch (err) {
+      let msg = `An Internal Error occurred in validating the spec "${self.specPath}". \t${err.message}.`;
+      err.code = ErrorCodes.InternalError;
+      err.message = msg;
+      self.specValidationResult.validateSpec.isValid = false;
+      self.specValidationResult.validateSpec.error = err;
+      log.error(err);
+      self.updateValidityStatus();
     }
+  }
 
-    return function (callback) {
-      swt.validateModel(self.specInJson, modelReference, data, callback);
+  getOperationById(id) {
+    let self = this;
+    if (!self.swaggerApi) {
+      throw new Error(`Please call specValidator.initialize() so that swaggerApi is populated, before calling this method.`);
+    }
+    if (!id) {
+      throw new Error(`id cannot be null or undefined and must be of type string.`);
     }
+    let result = this.swaggerApi.getOperations().find(function(item) {
+      return (item.operationId === id);
+    });
+    return result;
   }
 
-  validateDateTypes(schema, value) {
-    if (value !== null && value !== undefined) {
-      if (schema.format.match(/^date$/ig) !== null) {
-        if (!(value instanceof Date ||
-          (typeof value.valueOf() === 'string' && !isNaN(Date.parse(value))))) {
-          throw new Error(`${schema.name} must be an instanceof Date or a string in ISO8601 format.`);
-        }
-      } else if (schema.format.match(/^date-time$/ig) !== null) {
-        if (!(value instanceof Date ||
-          (typeof value.valueOf() === 'string' && !isNaN(Date.parse(value))))) {
-          throw new Error(`${schema.name} must be an instanceof Date or a string in ISO8601 format.`);
+  getXmsExamples(idOrObj) {
+    if (!idOrObj) {
+      throw new Error(`idOrObj cannot be null or undefined and must be of type string or object.`);
+    }
+    let operation = {};
+    if (typeof idOrObj.valueOf() === 'string') {
+      operation = self.getOperationById(id);
+    } else {
+      operation = idOrObj;
+    }
+    let result;
+    if (operation && operation[Constants.xmsExamples]) {
+      result = operation[Constants.xmsExamples];
+    }
+    return result;
+  }
+
+  constructOperationResult(operation, result, exampleType) {
```

Comment:
yup. want to refactor this and make it super small. Will do this later on after making sure all the P0 issues are resolved.